src/models/lazy_load.rs

```diff
@@ -599,6 +603,25 @@ impl<T: Clone + 'static> LazyItemMap<T> {
         .unwrap();
     }
 
+    /// Inserts a new key value pair into the map if the key does not already exist
+    ///
+    /// Note: Concurrent updates should use checked_insert to avoid overwriting
+    /// updates from other threads.
+    pub fn checked_insert(&self, key: IdentityMapKey, default: LazyItem<T>) {
```

Comment:
@a-rustacean @tinkn This prevents losing updates when doing concurrent writes. Please take a look.